src/main/webapp/app/shared/util/firebase/firebase-utils.ts

```diff
@@ -318,3 +334,136 @@ export const isMutationEffectCuratable = (mutationName: string) => {
   const excludedMutations = ['Oncogenic Mutations'];
   return excludedMutations.filter(mutation => mutationName.toLowerCase().includes(mutation.toLowerCase())).length === 0;
 };
+
+export function compareMutationsByDeleted(mut1: Mutation, mut2: Mutation) {
+  const mut1IsDeleted = mut1.name_review?.removed || false;
+  const mut2IsDeleted = mut2.name_review?.removed || false;
+
+  if ((mut1IsDeleted && mut2IsDeleted) || (!mut1IsDeleted && !mut2IsDeleted)) {
+    return 0;
+  } else if (mut1IsDeleted) {
+    return 1;
+  } else {
+    return -1;
+  }
+}
+
+export function compareMutationsByTxLevel(mut1: Mutation, mut2: Mutation, mutationLevelSummary: MutationLevelSummary) {
+  const mut1Levels = Object.keys(mutationLevelSummary[mut1.name_uuid]?.txLevels || []).sort(sortByTxLevel);
+  const mut2Levels = Object.keys(mutationLevelSummary[mut2.name_uuid]?.txLevels || []).sort(sortByTxLevel);
+
+  let index = 0;
+  while (mut1Levels[index] || mut2Levels[index]) {
+    if (!mut1Levels[index]) {
+      return 1;
+    } else if (!mut2Levels[index]) {
+      return -1;
+    }
+
+    const order = sortByTxLevel(mut1Levels[index] as TX_LEVELS, mut2Levels[index] as TX_LEVELS);
+    if (order !== 0) {
+      return order;
+    }
+
+    index++;
+  }
+
+  return 0;
+}
+
+export function compareMutationsByOncogenicity(mut1: Mutation, mut2: Mutation) {
```

Comment:
Visually since we gonna use the same blue icon for oncogenic/likely/resistance, it's a bit confusing sorting based on their priority then by position. Let's try the priority of oncogenic/likely/resistance the same.